File: 04_Frontend_Application/Edums-Frontend/src/app/modules/m1-school-office/interface/official-circular.ts
import type { CircularType, TargetAudience } from './common';

export interface OfficialCircular {
  id: number;
  circularNumber: string;
  issueDate: string;
  titleAr: string;
  titleEn: string;
  circularType: CircularType;
  issuerName: string;
  targetAudience: TargetAudience;
  effectiveDate: string;
  isActive: boolean;
  contentBody: string | null;
  issuerEmployeeId: number | null;
  attachmentFileUrl: string | null;
  requiresMandatoryAcknowledgment: boolean;
  acknowledgmentDeadline: string | null;
}

export type CreateOfficialCircularDto = Omit<OfficialCircular, 'id' | 'isActive' | 'effectiveDate'>;

export type UpdateOfficialCircularDto = Omit<OfficialCircular, 'isActive' | 'effectiveDate'>;

File: 01_Backend/EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/OfficialCircular.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class OfficialCircular : BaseAuditableEntity
{
    public string CircularNumber { get; set; } = string.Empty;
    public DateTime IssueDate { get; set; }
    public string TitleAr { get; set; } = string.Empty;
    public string TitleEn { get; set; } = string.Empty;
    public int CircularType { get; set; } // 1=Educational, 2=Financial, 3=Administrative
    public string IssuerName { get; set; } = string.Empty;
    public int TargetAudience { get; set; } // 1=AllSchools, 2=SpecificSchools, 3=Teachers
    public DateTime EffectiveDate { get; set; }
    public bool IsActive { get; set; } = true;
    public string? ContentBody { get; set; }
    public long? IssuerEmployeeId { get; set; }
    public string? AttachmentFileUrl { get; set; }
    public bool RequiresMandatoryAcknowledgment { get; set; }
    public DateTime? AcknowledgmentDeadline { get; set; }

    // Navigation Property
    public virtual Employee? IssuerEmployee { get; set; }
}
